controller/controller.go

```diff
@@ -141,6 +141,10 @@ func (c *Impl) EnqueueControllerOf(obj interface{}) {
 // converts it into a namespace/name string, and passes that to EnqueueKey.
 // Callers should pass in an empty string as namespace label key for obj
 // whose controller is of cluster-scoped resource.
+//
+// Deprecated: this will be removed once all of its upstream usage get removed.
+// Use "EnqueueLabelOfNamespaceScopedResource" or "EnqueueLabelOfClusterScopedResource"
+// instead.
```

Comment:
I meant just delegate to `EnqueueLabelOfNamespaceScopeResource` here.

Reply:
Let's go with #2, and rip the bandaid off, sorry for the churn.

Reply:
done